packages/dx-react-scheduler-demos/src/demo-sources/scheduler-resources/material-ui/basic.jsx

```diff
@@ -0,0 +1,159 @@
+import MenuItem from '@material-ui/core/MenuItem';
+import { withStyles } from '@material-ui/core/styles';
+
+const appointments = [{
+  title: 'Website Re-Design Plan',
+  startDate: new Date(2018, 5, 25, 12, 35),
+  endDate: new Date(2018, 5, 25, 15, 0),
+  id: 0,
+  members: [1, 3, 5],
+  location: 'Room 1',
+}, {
+  title: 'Book Flights to San Fran for Sales Trip',
+  startDate: new Date(2018, 5, 26, 12, 35),
+  endDate: new Date(2018, 5, 26, 15, 0),
+  id: 1,
+  members: [2, 4],
+  location: 'Room 2',
+}, {
+  title: 'Install New Router in Dev Room',
+  startDate: new Date(2018, 5, 27, 12, 35),
+  endDate: new Date(2018, 5, 27, 15, 0),
+  id: 2,
+  members: [3],
+  location: 'Room 3',
+}, {
+  title: 'Approve Personal Computer Upgrade Plan',
+  startDate: new Date(2018, 5, 28, 12, 35),
+  endDate: new Date(2018, 5, 28, 15, 0),
+  id: 3,
+  members: [4, 1],
+  location: 'Room 4',
+}, {
+  title: 'Final Budget Review',
+  startDate: new Date(2018, 5, 29, 12, 35),
+  endDate: new Date(2018, 5, 29, 15, 0),
+  id: 4,
+  members: [5, 1, 3],
+  location: 'Room 5',
+}];
+
+const styles = theme => ({
+  container: {
+    display: 'flex',
+    marginBottom: theme.spacing(2),
+    justifyContent: 'flex-end',
+  },
+  text: {
+    ...theme.typography.h6,
+    marginRight: theme.spacing(2),
+  },
+});
+
+const ResourceSwitcher = withStyles(styles, { name: 'ResourceSwitcher' })(
+  ({
+    mainResourceName, onChange, classes, resources,
+  }) => (
+    <div className={classes.container}>
+      <div className={classes.text}>
+        Main resource name:
+      </div>
+      <Select
+        value={mainResourceName}
+        onChange={e => onChange(e.target.value)}
+      >
+        {resources.map(resource => (
+          <MenuItem key={resource.fieldName} value={resource.fieldName}>
+            {resource.title}
+          </MenuItem>
+        ))}
+      </Select>
+    </div>
+  ),
+);
+
+export default class Demo extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: appointments,
+      mainResourceName: 'members',
+      resources: [
+        {
+          fieldName: 'location',
+          title: 'Location',
+          items: [
+            { id: 'Room 1', text: 'Room 1' },
+            { id: 'Room 2', text: 'Room 2' },
+            { id: 'Room 3', text: 'Room 3' },
+            { id: 'Room 4', text: 'Room 4' },
+            { id: 'Room 5', text: 'Room 5' },
+          ],
+        },
+        {
+          fieldName: 'members',
+          title: 'Members',
+          allowMultiple: true,
+          items: [
+            { id: 1, text: 'Andrew Glover' },
+            { id: 2, text: 'Arnie Schwartz' },
+            { id: 3, text: 'John Heart' },
+            { id: 4, text: 'Taylor Riley' },
+            { id: 5, text: 'Brad Farkus' },
+          ],
+        },
+      ],
+    };
+
+    this.changeMainResource = this.changeMainResource.bind(this);
+  }
+
+  changeMainResource(mainResourceName) {
```

Comment:
This function can be declared in constructor using fat arrow then binding is unnecessary